components/aws/sagemaker/common/_utils.py

```diff
@@ -471,11 +471,61 @@ def create_model_request(args):
 
     return request
 
+def endpoint_name_exists(client, endpoint_name):
+  endpoint_exists = False
+  if endpoint_name:
```

Comment:
When do you expect this condition to catch? As far as I can tell, it would only catch if `endpoint_name == ""`.